### ISSUE TITLE
Device not appearing in MQTT

### ISSUE THREAD

**Issue by multicm** [richardhopton/smartbed-mqtt#4]
I am getting the following errors from MQTT preventing the entities from being discovered: 

```
Logger: homeassistant.util.logging
Source: util/logging.py:159
First occurred: February 27, 2023 at 11:46:06 PM (112 occurrences)
Last logged: 12:31:24 AM

Exception in async_discover when dispatching 'mqtt_discovery_new_switch_mqtt': ({'name': 'Bedroom Safety Lights', 'unique_id': 'bedroom_safety_lights', 'device': {'identifiers': ['768675_351152'], 'name': 'Bedroom', 'manufacturer': None, 'model': 'STS40'}, 'availability_topic': 'sleeptracker/768675_351152/safety_lights/status', 'payload_available': 'online', 'payload_not_available': 'offline', 'state_topic': 'sleeptracker/768675_351152/safety_lights/state', 'command_topic': 'sleeptracker/768675_351152/safety_lights/command', 'platform': 'mqtt'},) Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/components/mqtt/mixins.py", line 334, in async_discover config: DiscoveryInfoType = discovery_schema(discovery_payload) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 272, in __call__ return self._compiled([], data) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 595, in validate_dict return base_validate(path, iteritems(data), out) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 433, in validate_mapping raise er.MultipleInvalid(errors) voluptuous.error.MultipleInvalid: string value is None for dictionary value @ data['device']['manufacturer']
Exception in async_discover when dispatching 'mqtt_discovery_new_sensor_mqtt': ({'name': 'Bedroom Temperature Sensor', 'unique_id': 'bedroom_temperature_sensor', 'device': {'identifiers': ['768675_351152'], 'name': 'Bedroom', 'manufacturer': None, 'model': 'STS40'}, 'availability_topic': 'sleeptracker/768675_351152/temperature_sensor/status', 'payload_available': 'online', 'payload_not_available': 'offline', 'state_topic': 'sleeptracker/768675_351152/temperature_sensor/state', 'value_template': "{{ value_json.value | default('') }}", 'json_attributes_topic': 'sleeptracker/768675_351152/temperature_sensor/state', 'state_class': 'measurement', 'unit_of_measurement': '°C', 'device_class': 'temperature', 'platform': 'mqtt'},) Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/components/mqtt/mixins.py", line 334, in async_discover config: DiscoveryInfoType = discovery_schema(discovery_payload) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 232, in __call__ return self._exec((Schema(val) for val in self.validators), v) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 355, in _exec raise e if self.msg is None else AllInvalid(self.msg, path=path) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 351, in _exec v = func(v) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 272, in __call__ return self._compiled([], data) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 818, in validate_callable return schema(data) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 272, in __call__ return self._compiled([], data) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 595, in validate_dict return base_validate(path, iteritems(data), out) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 433, in validate_mapping raise er.MultipleInvalid(errors) voluptuous.error.MultipleInvalid: string value is None for dictionary value @ data['device']['manufacturer']
Exception in async_discover when dispatching 'mqtt_discovery_new_sensor_mqtt': ({'name': 'Bedroom Humidity Sensor', 'unique_id': 'bedroom_humidity_sensor', 'device': {'identifiers': ['768675_351152'], 'name': 'Bedroom', 'manufacturer': None, 'model': 'STS40'}, 'availability_topic': 'sleeptracker/768675_351152/humidity_sensor/status', 'payload_available': 'online', 'payload_not_available': 'offline', 'state_topic': 'sleeptracker/768675_351152/humidity_sensor/state', 'value_template': "{{ value_json.value | default('') }}", 'json_attributes_topic': 'sleeptracker/768675_351152/humidity_sensor/state', 'state_class': 'measurement', 'unit_of_measurement': '%', 'device_class': 'humidity', 'platform': 'mqtt'},) Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/components/mqtt/mixins.py", line 334, in async_discover config: DiscoveryInfoType = discovery_schema(discovery_payload) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 232, in __call__ return self._exec((Schema(val) for val in self.validators), v) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 355, in _exec raise e if self.msg is None else AllInvalid(self.msg, path=path) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 351, in _exec v = func(v) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 272, in __call__ return self._compiled([], data) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 818, in validate_callable return schema(data) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 272, in __call__ return self._compiled([], data) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 595, in validate_dict return base_validate(path, iteritems(data), out) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 433, in validate_mapping raise er.MultipleInvalid(errors) voluptuous.error.MultipleInvalid: string value is None for dictionary value @ data['device']['manufacturer']
Exception in async_discover when dispatching 'mqtt_discovery_new_sensor_mqtt': ({'name': 'Bedroom CO2 Sensor', 'unique_id': 'bedroom_co2_sensor', 'device': {'identifiers': ['768675_351152'], 'name': 'Bedroom', 'manufacturer': None, 'model': 'STS40'}, 'availability_topic': 'sleeptracker/768675_351152/co2_sensor/status', 'payload_available': 'online', 'payload_not_available': 'offline', 'state_topic': 'sleeptracker/768675_351152/co2_sensor/state', 'value_template': "{{ value_json.value | default('') }}", 'json_attributes_topic': 'sleeptracker/768675_351152/co2_sensor/state', 'state_class': 'measurement', 'unit_of_measurement': 'ppm', 'device_class': 'carbon_dioxide', 'icon': 'mdi:molecule-co2', 'platform': 'mqtt'},) Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/components/mqtt/mixins.py", line 334, in async_discover config: DiscoveryInfoType = discovery_schema(discovery_payload) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 232, in __call__ return self._exec((Schema(val) for val in self.validators), v) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 355, in _exec raise e if self.msg is None else AllInvalid(self.msg, path=path) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 351, in _exec v = func(v) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 272, in __call__ return self._compiled([], data) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 818, in validate_callable return schema(data) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 272, in __call__ return self._compiled([], data) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 595, in validate_dict return base_validate(path, iteritems(data), out) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 433, in validate_mapping raise er.MultipleInvalid(errors) voluptuous.error.MultipleInvalid: string value is None for dictionary value @ data['device']['manufacturer']
Exception in async_discover when dispatching 'mqtt_discovery_new_sensor_mqtt': ({'name': 'Bedroom VOC Sensor', 'unique_id': 'bedroom_voc_sensor', 'device': {'identifiers': ['768675_351152'], 'name': 'Bedroom', 'manufacturer': None, 'model': 'STS40'}, 'availability_topic': 'sleeptracker/768675_351152/voc_sensor/status', 'payload_available': 'online', 'payload_not_available': 'offline', 'state_topic': 'sleeptracker/768675_351152/voc_sensor/state', 'value_template': "{{ value_json.value | default('') }}", 'json_attributes_topic': 'sleeptracker/768675_351152/voc_sensor/state', 'state_class': 'measurement', 'unit_of_measurement': 'ppb', 'device_class': 'volatile_organic_compounds', 'icon': 'mdi:cloud', 'platform': 'mqtt'},) Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/components/mqtt/mixins.py", line 334, in async_discover config: DiscoveryInfoType = discovery_schema(discovery_payload) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 232, in __call__ return self._exec((Schema(val) for val in self.validators), v) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 355, in _exec raise e if self.msg is None else AllInvalid(self.msg, path=path) File "/usr/local/lib/python3.10/site-packages/voluptuous/validators.py", line 351, in _exec v = func(v) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 272, in __call__ return self._compiled([], data) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 818, in validate_callable return schema(data) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 272, in __call__ return self._compiled([], data) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 595, in validate_dict return base_validate(path, iteritems(data), out) File "/usr/local/lib/python3.10/site-packages/voluptuous/schema_builder.py", line 433, in validate_mapping raise er.MultipleInvalid(errors) voluptuous.error.MultipleInvalid: string value is None for dictionary value @ data['device']['manufacturer']
```

**Comment by multicm** [comment]
Just to add I did try to Uninstaller both this add-on and mosquito, restart, and reinstall. But the same issue persists

**Comment by richardhopton** [comment]
Looks like the Sleeptracker API is not always returning a manufacturer, but HA needs one. I'll make a small change to fix

**Comment by richardhopton** [comment]
fixed in 1.0.1
